src/fix_pair.cpp

```diff
@@ -0,0 +1,343 @@
+// clang-format off
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "fix_pair.h"
+
+#include "atom.h"
+#include "dump.h"
+#include "error.h"
+#include "force.h"
+#include "fix.h"
+#include "input.h"
+#include "memory.h"
+#include "pair.h"
+#include "output.h"
+#include "variable.h"
+#include "update.h"
+#include "variable.h"
+
+#include <cstring>
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+FixPair::FixPair(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg)
+{
+  if (narg < 7) utils::missing_cmd_args(FLERR, "fix pair", error);
+
+  nevery = utils::inumeric(FLERR,arg[3],false,lmp);
+  if (nevery < 1) error->all(FLERR,"Illegal fix pair every value: {}", nevery);
+
+  pairname = utils::strdup(arg[4]);
+  pstyle = force->pair_match(pairname,1,0);
+  if (pstyle == nullptr) error->all(FLERR,"Pair style {} for fix pair not found", pairname);
+
+  nfield = (narg-5) / 2;
+  fieldname = new char*[nfield];
+  trigger = new int[nfield];
+
+  nfield = 0;
+  int iarg = 5;
+
+  while (iarg < narg) {
+    if (iarg+2 > narg) utils::missing_cmd_args(FLERR, fmt::format("fix pair {}", arg[iarg]), error);
+    fieldname[nfield] = utils::strdup(arg[iarg]);
+    int flag = utils::inumeric(FLERR,arg[iarg+1],true,lmp);
+    if (flag == 0) trigger[nfield] = 0;
+    else if (flag == 1) trigger[nfield] = 1;
+    else error->all(FLERR,"Illegal fix pair {} command flag: {}", arg[iarg], arg[iarg+1]);
+    nfield++;
+    iarg += 2;
+  }
+
+  // set trigger names = fieldname + "_flag"
+
+  triggername = new char*[nfield];
+
+  for (int ifield = 0; ifield < nfield; ifield++) {
+    if (trigger[ifield] == 0) triggername[ifield] = nullptr;
+    else triggername[nfield] = utils::strdup(fmt::format("{}_flag", fieldname[ifield]));
```

Comment:
I think it should be "else triggername[ifield] = "

Reply:
Thanks. I agree. Fix will be added to a different pull request, since I merged this one 15 minutes ago.

Reply:
https://github.com/lammps/lammps/pull/3444/commits/3770c02752bddfc26ee8c1385c84867f58781a02